src/stan/model/indexing/index.hpp

```diff
@@ -87,15 +87,16 @@ struct index_max {
 struct index_min_max {
   int min_;
   int max_;
-
+  bool positive_idx_{true};  // If true min <= max
```

Comment:
Why do you need this in struct? You can always do the comparison where do you need it. Or do you need it in many places, so it makes sense to do it beforehand?

Reply:
tbh I had I was doing it in so many places I put it here. It made the code looks a lot nicer. But thinking about it I could / should turn this just into a member function or pull it out